submissions/Levi123/memory-pair-game/script.js

```diff
@@ -0,0 +1,142 @@
+const SELECTORS = {
+    flipContainer : document.querySelector('.flip-container'),
+    content : document.querySelector('.main-content__playing-field'),
+    textContent: document.querySelector('.main-content'),
+    timer : document.querySelector('.seconds'),
+    flipCounter: document.querySelector('.flips'),
+}
+
+const STATE = {
+    totalFlips : 0,
+    totalTime : 0,
+    flipElementIdArray : [],
+    cardOne : undefined,
+    cardTwo : undefined,
+    statusGame : false,
+    loop: null,
+    combinationCounter: 6,
+    isInteractive: true,
+}
+
+const generateGame = () => {
+    const allElementsForGame = [
+        {
+            id:1,
+            images: 'img/bear.png'
+        },
+    
+        {
+            id:2,
+            images: 'img/elephant.png'
+        },
+    
+        {
+            id:3,
+            images: 'img/giraffe.png'
+        },
+    
+        {
+            id:4,
+            images: 'img/grizzly.png'
+        },
+    
+        {
+            id:5,
+            images: 'img/hippo.png'
+        },
+    
+        {
+            id:6,
+            images: 'img/koala.png'
+        },
+    ]
+
+
+    function shuffle(arrayToShuffle) {
+        for(var j, x, i = arrayToShuffle.length; i; j = parseInt(Math.random() * i), x = arrayToShuffle[--i], arrayToShuffle[i] = arrayToShuffle[j], arrayToShuffle[j] = x);
+        return arrayToShuffle;
+    };
+
+    const generateElementForBoard = shuffle([...allElementsForGame, ...allElementsForGame]);
+    console.log(generateElementForBoard);
+
+    for (let i of generateElementForBoard){
+        let list = `
+            <div class="main-content__element flipper">
+                <div class="front">
+                    <img src="img/card.png" alt="" class="flip-card-img" data-id=${i.id}>
+                </div>
+                <div class="back">
+                    <img src=${i.images} alt="" class="flip-card-back-img">
+                </div>
+            </div>
+        `
```

Comment:
Try methods `reduce` or `map`

Reply:
@madmaxWMFU used the forEach method, will it be appropriate?